### DISCUSSION TITLE
Sections in Assert.Multiple

### DISCUSSION THREAD

**Discussion by VulumeCode** [nunit/nunit#4742] [Q&A]
Is there a cleaner way to achieve this? 

````
            Assert.Multiple(() =>
            {
                Assert.DoesNotThrow(()=>
                {
                    Assert.Fail("A");
                    Assert.Fail("Don't want to see this");
                });
                Assert.DoesNotThrow(()=>
                {
                    throw new Exception("B");
                });
                Assert.DoesNotThrow(()=>
                {
                    Assert.Fail("C");
                    Assert.Fail("Don't want to see this");
                });
            });
````

So that it just shows the 3 failures (A, B, C) with stack traces to the failure points, without all the verbosity about the thrown exception? 

If not, would this be a desired feature? What would it look like?

**Comment by stevenaw** [comment]
@VulumeCode 
Just to confirm: Are you looking for a way to group assertions inside of a "multiple" block so that each subgroup will fail-fast and only report the first failure?

Off the top of my head no way currently exists, though some ideas have been proposed recently which could allow for this (#4461 and https://github.com/nunit/nunit/issues/4587#issuecomment-1856490329) however nothing has been decided by the team as of yet

**Comment by manfred-brands** [comment]
@VulumeCode Can you explain why you need this instead of making separate tests?

NUnit runs all test, regardless of failure of the others.
You don't need to add `DoesNotThrow` as that is implied.

```
[OneTimeSetUp]
public void CommonSetUp()
{
    ....
}

[Test]
public void Test1()
{
    Assert.Fail("A");
    Assert.Fail("Don't want to see this");
}

[Test]
public void Test2()
{
    throw new Exception("B");
}

[Test]
public void Test3()
{
     Assert.Fail("C");
     Assert.Fail("Don't want to see this");
}

**Comment by VulumeCode** [comment]
@manfred-brands It's for longer running scenarios, with lots of setup and changes to the setup in between "Asserts.Multiple"s which can actually fail the whole test. 

I tried it with nested classes with [OneTimeSetUp]s and sequencing tests etc but it gets very verbose and confusing.
